### ISSUE TITLE
compareInt2LargeSIMD can't be used without cutoff

### ISSUE THREAD

**Issue by ex-rzr** [openmm/openmm#3462]
https://github.com/openmm/openmm/blob/master/platforms/opencl/src/OpenCLNonbondedUtilities.cpp#L230

The OpenCL platform uses a different order of tiles with exclusions on AMD GPUs with warps of 64 (GCN, CDNA). It groups diagonal tiles in `exclusionTiles` together to eliminate cases when one half of a wrap processes a diagonal tile and another precesses an off-diagonal tile, improving divergence in the first loop of kernels like nonbonded.

However, `exclusionTiles` is also used later in these kernels: if interactions don't use cutoff, the kernel needs to process all tiles except the tiles with exclusions processed in the first loop.

The code doing it depends on a correct order of tiles in `exclusionTiles`:
https://github.com/openmm/openmm/blob/master/platforms/opencl/src/kernels/nonbonded.cl#L245-L259

`compareInt2` really sorts all tiles so this code works correctly, but `compareInt2LargeSIMD` basically splits the list in two (diagonal and off-diaonal tiles) and sorts them independently.

So there are situations when some tiles are processed twice: as a tile with exclusions and as a tile without exclusions (without considering that some interactions within the tile must be excluded).

I added this piece of code to all files where such filtering with skipTiles is used:

```cpp
        // after includeTile = (skipTiles[currentSkipIndex] != pos);

        if (get_local_id(0) % 32 == 0) {
            bool found = false;
            for (int ii = 0; ii < NUM_TILES_WITH_EXCLUSIONS; ii++) {
                int2 tile = exclusionTiles[ii];
                int p = tile.x + tile.y*NUM_BLOCKS - tile.y*(tile.y+1)/2;
                if (pos == p) {
                    found = true;
                }
            }
            if (includeTile) {
                if (found) {
                    printf("!!! incorrectly included %d (%d, %d)\n", pos, x, y);
                }
            }
            else {
                if (!found) {
                    printf("!!! incorrectly excluded %d (%d, %d)\n", pos, x, y);
                }
            }
        }
```

There are 13 such files (.cc and .cl). I ran OpenCL tests on AMD GPU (ROCm OpenCL), "!!! incorrectly included" is reported for several tiles by 3 tests:
* TestOpenCLAmoebaGeneralizedKirkwoodForce
* TestOpenCLCustomNonbondedForce
* TestOpenCLNonbondedForce

and one benchmark: amoebagk.

I guess some tests may be affected by the issue but too small to trigger my check (like when there is just one block).

Were there simulations when OpenCL on GCN/CDNA GPUs produced incorrect results?

**Comment by peastman** [comment]
Good catch!  You're right, we should force it to use `compareInt2` when there's no cutoff.  Fortunately, that's a very rare situation except for very small systems.

**Comment by peastman** [comment]
The fix is in #3464.  Can you verify that it's correct?

**Comment by ex-rzr** [comment]
The tests work correctly, thanks!
